spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jdbc/DataSourceAutoConfiguration.java

```diff
@@ -67,10 +69,19 @@
 	private static final Log logger = LogFactory
 			.getLog(DataSourceAutoConfiguration.class);
 
+	private final DataSourceProperties properties;
+
+	private final ApplicationContext applicationContext;
+
+	public DataSourceAutoConfiguration(DataSourceProperties properties, ApplicationContext applicationContext) {
```

Comment:
@wilkinsona, I agree with your suggestion. I've fixed this.
Please review again !